From f6a7fb78668949e5d8983bd23086f6f37efda08a Mon Sep 17 00:00:00 2001
From: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
Date: Wed, 27 May 2020 20:02:15 +0530
Subject: [PATCH 9/9] pcie-pinctrl-modifications

- This patch will do pinctrl and respective gpio modifications for pcie0 and pcie1.
- Removed sai1 node, respective pinctrl and related audio codecs ak4458, ak4497.
- Removed conflicting qspi0 node and its pinctrl.

Signed-off-by: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
---
 .../boot/dts/adlink/adlink-lec-imx8m.dts      | 126 +-----------------
 arch/arm64/configs/lec_imx8m_defconfig        |   2 -
 2 files changed, 4 insertions(+), 124 deletions(-)

diff --git a/arch/arm64/boot/dts/adlink/adlink-lec-imx8m.dts b/arch/arm64/boot/dts/adlink/adlink-lec-imx8m.dts
index 24370f83bce3..99de50e65403 100755
--- a/arch/arm64/boot/dts/adlink/adlink-lec-imx8m.dts
+++ b/arch/arm64/boot/dts/adlink/adlink-lec-imx8m.dts
@@ -148,28 +148,12 @@
 				<192000>;
 	};
 
-	sound-ak4458 {
-		compatible = "fsl,imx-audio-ak4458-mq";
-		model = "ak4458-audio";
-		audio-cpu = <&sai1>;
-		audio-codec = <&ak4458_1>, <&ak4458_2>;
-		ak4458,pdn-gpio = <&gpio3 18 GPIO_ACTIVE_HIGH>;
-	};
-
 	sound-ak5558 {
 		compatible = "fsl,imx-audio-ak5558-mq";
 		model = "ak5558-audio";
 		audio-cpu = <&sai5>;
 		audio-codec = <&ak5558>;
 	};
-
-	sound-ak4497 {
-		compatible = "fsl,imx-audio-ak4497-mq";
-		model = "ak4497-audio";
-		audio-cpu = <&sai1>;
-		audio-codec = <&ak4497>;
-		status = "disabled";
-	};
 };
 
 &CPU_SLEEP {
@@ -317,25 +301,6 @@
 	status = "okay";
 };
 
-&sai1 {
-	pinctrl-names = "default", "pcm_b2m", "dsd";
-	pinctrl-0 = <&pinctrl_sai1_pcm>;
-	pinctrl-1 = <&pinctrl_sai1_pcm_b2m>;
-	pinctrl-2 = <&pinctrl_sai1_dsd>;
-	assigned-clocks = <&clk IMX8MQ_CLK_SAI1>;
-	assigned-clock-parents = <&clk IMX8MQ_AUDIO_PLL1_OUT>;
-	assigned-clock-rates = <49152000>;
-	clocks = <&clk IMX8MQ_CLK_SAI1_IPG>, <&clk IMX8MQ_CLK_DUMMY>,
-		<&clk IMX8MQ_CLK_SAI1_ROOT>, <&clk IMX8MQ_CLK_DUMMY>,
-		<&clk IMX8MQ_CLK_DUMMY>, <&clk IMX8MQ_AUDIO_PLL1_OUT>,
-		<&clk IMX8MQ_AUDIO_PLL2_OUT>;
-	clock-names = "bus", "mclk0", "mclk1", "mclk2", "mclk3", "pll8k", "pll11k";
-	fsl,sai-multi-lane;
-	fsl,dataline,dsd = <0 0xff 0xff 2 0xff 0x11>;
-	dmas = <&sdma2 8 25 0>, <&sdma2 9 25 0>;
-	status = "okay";
-};
-
 &sai4 {
 	assigned-clocks = <&clk IMX8MQ_CLK_SAI4>;
 	assigned-clock-parents = <&clk IMX8MQ_AUDIO_PLL1_OUT>;
@@ -601,7 +566,7 @@
 &pcie0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_pcie0>;
-	reset-gpio = <&gpio5 28 GPIO_ACTIVE_LOW>;
+	reset-gpio = <&gpio3 0 GPIO_ACTIVE_LOW>;
 	clocks = <&clk IMX8MQ_CLK_PCIE1_ROOT>,
 		 <&clk IMX8MQ_CLK_PCIE1_AUX>,
 		 <&clk IMX8MQ_CLK_PCIE1_PHY>,
@@ -615,7 +580,7 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_pcie1>;
 	disable-gpio = <&gpio5 10 GPIO_ACTIVE_LOW>;
-	reset-gpio = <&gpio5 12 GPIO_ACTIVE_LOW>;
+	reset-gpio = <&gpio4 11 GPIO_ACTIVE_LOW>;
 	clocks = <&clk IMX8MQ_CLK_PCIE2_ROOT>,
 		 <&clk IMX8MQ_CLK_PCIE2_AUX>,
 		 <&clk IMX8MQ_CLK_PCIE2_PHY>,
@@ -637,23 +602,6 @@
 	status = "okay";
 };
 
-&qspi0 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_qspi>;
-	status = "okay";
-
-	flash0: n25q256a@0 {
-		reg = <0>;
-		#address-cells = <1>;
-		#size-cells = <1>;
-		compatible = "jedec,spi-nor";
-		spi-max-frequency = <29000000>;
-		spi-tx-bus-width = <4>;
-		spi-rx-bus-width = <4>;
-		spi-nor,ddr-quad-read-dummy = <6>;
-	};
-};
-
 &uart1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart1>;
@@ -832,28 +780,13 @@
 
 	pinctrl_pcie0: pcie0grp {
 		fsl,pins = <
-			MX8MQ_IOMUXC_I2C4_SCL_PCIE1_CLKREQ_B		0x76
-			MX8MQ_IOMUXC_UART4_RXD_GPIO5_IO28		0x16
+			MX8MQ_IOMUXC_NAND_ALE_GPIO3_IO0                 0x16
 		>;
 	};
 
 	pinctrl_pcie1: pcie1grp {
 		fsl,pins = <
-			MX8MQ_IOMUXC_I2C4_SDA_PCIE2_CLKREQ_B		0x76 /* open drain, pull up */
-			MX8MQ_IOMUXC_ECSPI2_SCLK_GPIO5_IO10		0x16
-			MX8MQ_IOMUXC_ECSPI2_MISO_GPIO5_IO12		0x16
-		>;
-	};
-
-	pinctrl_qspi: qspigrp {
-		fsl,pins = <
-			MX8MQ_IOMUXC_NAND_ALE_QSPI_A_SCLK	0x82
-			MX8MQ_IOMUXC_NAND_CE0_B_QSPI_A_SS0_B	0x82
-			MX8MQ_IOMUXC_NAND_DATA00_QSPI_A_DATA0	0x82
-			MX8MQ_IOMUXC_NAND_DATA01_QSPI_A_DATA1	0x82
-			MX8MQ_IOMUXC_NAND_DATA02_QSPI_A_DATA2	0x82
-			MX8MQ_IOMUXC_NAND_DATA03_QSPI_A_DATA3	0x82
-
+			MX8MQ_IOMUXC_SAI1_TXC_GPIO4_IO11                0x16
 		>;
 	};
 
@@ -875,57 +808,6 @@
 		>;
 	};
 
-	pinctrl_sai1_pcm: sai1grp_pcm {
-		fsl,pins = <
-			MX8MQ_IOMUXC_SAI1_MCLK_SAI1_MCLK		0xd6
-			MX8MQ_IOMUXC_SAI1_TXFS_SAI1_TX_SYNC		0xd6
-			MX8MQ_IOMUXC_SAI1_RXD7_SAI1_TX_SYNC		0xd6
-			MX8MQ_IOMUXC_SAI1_TXC_SAI1_TX_BCLK		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD0_SAI1_TX_DATA0		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD1_SAI1_TX_DATA1		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD2_SAI1_TX_DATA2		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD3_SAI1_TX_DATA3		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD4_SAI1_TX_DATA4		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD5_SAI1_TX_DATA5		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD6_SAI1_TX_DATA6		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD7_SAI1_TX_DATA7		0xd6
-		>;
-	};
-
-	pinctrl_sai1_pcm_b2m: sai1grp_pcm_b2m {
-		fsl,pins = <
-			MX8MQ_IOMUXC_SAI1_MCLK_SAI1_TX_BCLK		0xd6
-			MX8MQ_IOMUXC_SAI1_TXFS_SAI1_TX_SYNC		0xd6
-			MX8MQ_IOMUXC_SAI1_RXD7_SAI1_TX_SYNC		0xd6
-			MX8MQ_IOMUXC_SAI1_TXC_SAI1_TX_BCLK		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD0_SAI1_TX_DATA0		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD1_SAI1_TX_DATA1		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD2_SAI1_TX_DATA2		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD3_SAI1_TX_DATA3		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD4_SAI1_TX_DATA4		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD5_SAI1_TX_DATA5		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD6_SAI1_TX_DATA6		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD7_SAI1_TX_DATA7		0xd6
-		>;
-	};
-
-	pinctrl_sai1_dsd: sai1grp_dsd {
-		fsl,pins = <
-			MX8MQ_IOMUXC_SAI1_MCLK_SAI1_MCLK		0xd6
-			MX8MQ_IOMUXC_SAI1_TXFS_SAI1_TX_SYNC		0xd6
-			MX8MQ_IOMUXC_SAI1_RXD7_SAI1_TX_DATA4		0xd6
-			MX8MQ_IOMUXC_SAI1_TXC_SAI1_TX_BCLK		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD0_SAI1_TX_DATA0		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD1_SAI1_TX_DATA1		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD2_SAI1_TX_DATA2		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD3_SAI1_TX_DATA3		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD4_SAI1_TX_DATA4		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD5_SAI1_TX_DATA5		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD6_SAI1_TX_DATA6		0xd6
-			MX8MQ_IOMUXC_SAI1_TXD7_SAI1_TX_DATA7		0xd6
-		>;
-	};
-
 	pinctrl_sai5: sai5grp {
 		fsl,pins = <
 			MX8MQ_IOMUXC_SAI5_MCLK_SAI5_MCLK	0xd6
diff --git a/arch/arm64/configs/lec_imx8m_defconfig b/arch/arm64/configs/lec_imx8m_defconfig
index 4b37fe166139..1bcdf7ebb153 100644
--- a/arch/arm64/configs/lec_imx8m_defconfig
+++ b/arch/arm64/configs/lec_imx8m_defconfig
@@ -503,9 +503,7 @@ CONFIG_SND_SOC=y
 CONFIG_SND_SOC_FSL_DAI=m
 CONFIG_SND_SOC_FSL_EASRC=y
 CONFIG_SND_IMX_SOC=y
-CONFIG_SND_SOC_IMX_AK4458=y
 CONFIG_SND_SOC_IMX_AK5558=y
-CONFIG_SND_SOC_IMX_AK4497=y
 CONFIG_SND_SOC_IMX_WM8960=y
 CONFIG_SND_SOC_IMX_WM8524=y
 CONFIG_SND_SOC_IMX_CS42888=y
-- 
2.17.1

